src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -246,14 +292,15 @@ public void generateGeodatabase(Envelope extent) {
 
       // create generate geodatabase parameters for the current extent
       final ListenableFuture<GenerateGeodatabaseParameters> defaultParameters = geodatabaseSyncTask
-              .createDefaultGenerateGeodatabaseParametersAsync(extent);
+              .createDefaultGenerateGeodatabaseParametersAsync(geodatabaseExtentGraphics.getGeometry());
       defaultParameters.addDoneListener(() -> {
         try {
           // set parameters and don't include attachments
           GenerateGeodatabaseParameters parameters = defaultParameters.get();
           parameters.setReturnAttachments(false);
 
           // create a temporary file for the geodatabase
+          final AtomicInteger replica = new AtomicInteger();
```

Comment:
Done. Don't need to disable it again since it's already disabled right after clicking it, before `generateGeodatabase()` it's called. So I've changed this function to not reactivate the button. It gets reactivated once a feature is moved, like you suggested. Looks like I had it activate twice before that, actually (once after generating the geoDB and once after moving a feature).